ui/packages/catalog-ui-search/src/main/webapp/component/result-form/result-form.collection.js

```diff
@@ -17,19 +17,45 @@
  const ResultForm = require('component/search-form/search-form')
  const Common = require('js/Common')
  const user = require('component/singletons/user-instance')
- const Results = require('component/result-form/result-form')
- const wreqr = require('wreqr')
+ const $ = require('jquery')
+ const _ = require('underscore')
+
+ let resultTemplates = [];
+ let promiseIsResolved = false;
+ const resultTemplatePromise = () => $.ajax({
+     type: 'GET',
+     url: '/search/catalog/internal/forms/result',
+     contentType: 'application/json',
+     success: function(data) {
+         //Find templates with the same id but different property maps (because we should trust the server)
+         let updatedTemplates = data.filter(
+             incomingTemplate => _.any(resultTemplates, (cachedTemplate) => cachedTemplate.id === incomingTemplate.id && !_.isEqual(cachedTemplate, incomingTemplate))
+         );
+         //Find templates that are new
+         let newTemplates = data.filter(
+             incomingTemplate => resultTemplates.length === 0 || !_.any(resultTemplates, (cachedTemplate) => cachedTemplate.id === incomingTemplate.id)
+         );
+         //Replace updated templates in their corresponding indices (//TODO: Should this just be a backbone collection instead of an array?)
```

Comment:
Address todo?

Reply:
The Search form CRUD PR already addressed the question and left it the way it is implemented.  Removing TODO statement